plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,605 @@
+                                               .add_dim(0, N)
+                                               .add_dim(0, (OFF_C / DG) / 2)
+                                               .add_dim(off_h, off_h + 1)
+                                               .add_dim(off_w, off_w + 1);
+            single_offset_h = op::reshape(single_offset_h, make_tuple<int64_t>({N, 1, F_H, F_W}));
+            single_offset_w = op::reshape(single_offset_w, make_tuple<int64_t>({N, 1, F_H, F_W}));
+            if (offset_h_concat_axis_3.rank() != 4) {
+              offset_h_concat_axis_3 = single_offset_h;
+            } else {
+              offset_h_concat_axis_3 = op::concatenate({offset_h_concat_axis_3, single_offset_h}, 3);
+            }
+            if (offset_w_concat_axis_3.rank() != 4) {
+              offset_w_concat_axis_3 = single_offset_w;
+            } else {
+              offset_w_concat_axis_3 = op::concatenate({offset_w_concat_axis_3, single_offset_w}, 3);
+            }
+          }
+          if (offset_h_concat_axis_2.rank() != 4) {
+            offset_h_concat_axis_2 = offset_h_concat_axis_3;
+          } else {
+            offset_h_concat_axis_2 = op::concatenate({offset_h_concat_axis_2, offset_h_concat_axis_3}, 2);
+          }
+          if (offset_w_concat_axis_2.rank() != 4) {
+            offset_w_concat_axis_2 = offset_w_concat_axis_3;
+          } else {
+            offset_w_concat_axis_2 = op::concatenate({offset_w_concat_axis_2, offset_w_concat_axis_3}, 2);
+          }
+        }
+        offset_h_slices.push_back(offset_h_concat_axis_2);
+        offset_w_slices.push_back(offset_w_concat_axis_2);
+      }
+      // Define new height and width.
+      auto NEW_H = F_H * OFF_H, NEW_W = F_W * OFF_W;
+      // Broadcast offset.
+      edsl::Tensor offset_h(0), offset_w(0);
+      for (auto dg = 0; dg < DG; ++dg) {
+        edsl::Tensor offset_h_dg = offset_h_slices[dg];
+        edsl::Tensor offset_w_dg = offset_w_slices[dg];
+        offset_h_dg = op::broadcast(
+            offset_h_dg,
+            {static_cast<int>(N), static_cast<int>(CI / DG), static_cast<int>(NEW_H), static_cast<int>(NEW_W)},
+            {0, 1, 2, 3});
+        offset_w_dg = op::broadcast(
+            offset_w_dg,
+            {static_cast<int>(N), static_cast<int>(CI / DG), static_cast<int>(NEW_H), static_cast<int>(NEW_W)},
+            {0, 1, 2, 3});
+        if (offset_h.rank() != 4) {
+          offset_h = offset_h_dg;
+        } else {
+          offset_h = op::concatenate({offset_h, offset_h_dg}, 1);
+        }
+        if (offset_w.rank() != 4) {
+          offset_w = offset_w_dg;
+        } else {
+          offset_w = op::concatenate({offset_w, offset_w_dg}, 1);
+        }
+      }
+      // Define height index of input.
+      TensorShape shape_h(precision, {NEW_H, NEW_W});
+      Buffer buffer_h(shape_h);
+      std::vector<float> data_h;
+      for (auto off_h = 0; off_h < OFF_H; ++off_h) {
```

Comment:
Can we use `edsl::index` instead of the nested for loops and `copy_from` function ?